### PR TITLE
add bond order assignment and ring status assignment to atomic graphs

### PR DESCRIPTION
**Pull request by a-r-j**
This one took some thinking! I've added a couple edge functions for atomic graphs to assign bond orders and ring status to the edges. No checking of aromaticity etc, my O-chem knowledge was hard won and all too easily lost :p 

You can see the remnant of my original approach commented out at the bottom. In the end, I decided to hope that bond lengths are sufficient for determining bond order for our purposes. This assumption may break at some point, but I think it's a compelling heuristic and developing an even more principled approach is likely a time sink right now (turns out it's actually a non-trivial problem!)

### CODE REVIEW

**Comment by a-r-j** [comment]
Hey @ericmjl, I've been trying to get this tutorial up and running with Binder. I think the environment etc builds properly but I keep getting errors relating to the imports within graphein (hence the slew of commits). I've been trying to understand this but I really can't make heads or tails from this from what I've read or by looking at other repos. 

Example:

```python
from graphein.protein.config import ProteinGraphConfig
```

results in:

```
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-2-427fb779643b> in <module>
----> 1 from graphein.protein.config import ProteinGraphConfig
      2 
      3 config = ProteinGraphConfig()
      4 config.dict()

/srv/conda/envs/notebook/lib/python3.8/site-packages/graphein/__init__.py in <module>
      3 # License: BSD 3 clause
      4 # Code Repository: https://github.com/a-r-j/graphein
----> 5 from .protein import *
      6 from .rna import *
      7 from .utils import *

/srv/conda/envs/notebook/lib/python3.8/site-packages/graphein/protein/__init__.py in <module>
----> 1 from graphs import *
      2 from meshes import *
      3 from resi_atoms import *
      4 
      5 from utils import *

ModuleNotFoundError: No module named 'graphs'
```

**Comment by ericmjl** [comment]
Oh man, I totally let this one slip... my apologies! If my memory of Python packaging serves me right, the `MANIFEST.in` is not necessary. If you put an `__init__.py` file in a subdirectory, then that subdirectory will become importable from the bigger package. (The `__init__.py` is a "magic" file.) It's also the place where the top-level imports of a module, or submodule, can live.

**Comment by ericmjl** [comment]
Oh man, I totally let this one slip... my apologies! If my memory of Python packaging serves me right, the `MANIFEST.in` is not necessary. If you put an `__init__.py` file in a subdirectory, then that subdirectory will become importable from the bigger package. (The `__init__.py` is a "magic" file.) It's also the place where the top-level imports of a module, or submodule, can live.

File: graphein/protein/__init__.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from .config import *
+from .edges import *
+from .features import *
+from .graphs import *
+from .resi_atoms import *
+from .utils import *
```
Comment (ericmjl) [inline graphein/protein/__init__.py:6]: The `*` imports might not be the most ideal. I think the `__init__.py` is a great opportunity to define what exactly should be made high-level for a user. For example, in `protein`, I think `construct_graph` is a good one for the high-level API, as well as the relevant Config object. What do you think?

File: graphein/protein/resi_atoms.py [hunk 2]
```diff
@@ -1090,3 +1090,96 @@
         "NH2": "Nres",
     },
 }
+
+# Contains idealised single, double and triple bond lengths (i_s, i_d, i_t) and watersheds (w_sd, w_dt),
+# below which a bond is probably double/triple (e.g. triple < double < single)
+# Taken from: Automatic Assignment of Chemical Connectivity to Organic Molecules in the Cambridge Structural Database
```
Comment (ericmjl) [inline graphein/protein/resi_atoms.py:1096]: Love it! :tada:

File: graphein/utils.py [hunk 3]
```diff
@@ -57,6 +57,18 @@ def annotate_node_metadata(G: nx.Graph, funcs: List[Callable]) -> nx.Graph:
     return G
 
 
+def annotate_node_features(G: nx.Graph, funcs: List[Callable]) -> nx.Graph:
```
Comment (ericmjl) [inline graphein/utils.py:60]: Woohoo, properly refactored!

**Comment by ericmjl** [comment]
@a-r-j letting you know that I've looked at the changes. Most of them look good to me. It's about bedtime here, and I haven't had a chance to carefully look at the specifics of bond order assignment and ring status assignment, but I'll get to it tomorrow.

Meanwhile, one other little thing - I'd encourage you to strip out the outputs from notebooks before committing them. This will help save space on the repository. (Some of my bigger ones, like Network Analysis Made Simple, were bloated because I didn't know about this practice beforehand.) You can use the [pre-commit](https://pre-commit.com) framework to manage this piece of the workflow with `nbstripout`, so all of that can be automated.

File: graphein/protein/__init__.py (hunk 1 above)
Comment (a-r-j) [reply graphein/protein/__init__.py:6]: I think you're right on this. Will keep this for now - I'm rather keen to consolidate some of the outstanding PRs. You've raised an issue on specifying what goes into each namespace more carefully. Perhaps this is best done after we've established the remaining functionality?

Comment (ericmjl) [reply graphein/protein/__init__.py:6]: Yep, that sounds about right to me.

**Comment by ericmjl** [comment]
Hey @a-r-j! Apologies for returning late here. I think you might need to make sure that every submodule has an `__init__.py`, and that the imports are made relative (i.e. `import .graphs` rather than `import graphs`).

**Pull request merged**